sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -819,6 +820,9 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
       redactMapString(map.asCaseSensitiveMap().asScala, maxFields)
     case map: Map[_, _] =>
       redactMapString(map, maxFields)
+    case t: TableSpec =>
+      redactMapString(t.options, maxFields)
+      redactMapString(t.properties, maxFields)
```

Comment:
I think we need to return a copy of `TableSpec` with options and properties redacted.

Reply:
Sorry I didn't get this right. Fixed.